### ISSUE TITLE
How to get the list of generated source files by the compiler?

### ISSUE THREAD

**Issue by pmlopes** [casid/jte#51]
Hi,

I'm trying to quickly integrate the compiler into a non-maven project, and I am doing:

```java
    TemplateEngine templateEngine = TemplateEngine.create(
        new DirectoryCodeResolver(source), target, ContentType.valueOf(contentType));
    templateEngine.setTrimControlStructures(trimControlStructures);
    templateEngine.setHtmlTags(htmlTags);
    templateEngine.setHtmlAttributes(htmlAttributes);
    templateEngine.setHtmlCommentsPreserved(htmlCommentsPreserved);

    int amount;

    try {
      templateEngine.cleanAll();
      amount = templateEngine.generateAll();
      // here instead of amount I'd like to have a List<String> with the generated file names
    } catch (Exception e) {
      fatal("Failed to generate templates: " + e.getMessage());
      return;
    }
```

As you can see it is a shameful copy and paste of the maven plugin. Hovever, now I'd like to get the list of generated java files instead of the number of generated files.

The reason is that I want to further process the generated files. This would be for a tight integration between jte, vertx and es4x (graaljs).

**Comment by casid** [comment]
@pmlopes wow, that sounds very intersting!

I put something together during lunch break, let me know if this works for this usecase: https://github.com/casid/jte/commit/8bdc6d359e954d74b4afaf573be4ce5798bccb85

**Comment by pmlopes** [comment]
Yes this is great as I can quickly automate it to be used outside a "maven/gradle" world as well as use it on a different language using graalvm/js!

**Comment by casid** [comment]
@pmlopes do you need a release, or is the snapshot good enough for you right now?

**Comment by pmlopes** [comment]
For now I can work with snapshots, as I'm still gluing stuff :) once I've it working I'll ping you

**Comment by casid** [comment]
I'm going to release jte 1.7.0 soon, this will be part of that release.

**Comment by casid** [comment]
Released with 1.7.0 👍
